### PR TITLE
feat(shared-data): add H/S to deck definition, bump deck definition schema

### PR DESCRIPTION
**Pull request by sanni-t**
# Overview

Adds heater-shaker module compatibility to deck definitions in anticipation of #9642.
Bumps the schema to version 3.

# Changelog
 
- added h/s module compatibility to slots for all deck types
- removed schema 1 & 2 definitions and fixtures
- updated api to use v3 deck, updated tests
- updated app, components & PD to use the new deck schema and definitions (by @shlokamin)

# Review requests

- Diff looks kinda big but it's mostly because of V1 & V2 deck removal. A good place to start reviewing would be `shared-data`. Everything else is basically switching all systems over to using V3. 
- Check that no unexpected changes have gone in, like removal of a field in a v3 deck definition
- Check that the correct slots on the deck are made compatible for heater-shaker (that's all except the trash). Especially check for short trash and ot3 because I am not familiar with those decks.
- All backend + FE/ client changes look good

# Risk assessment

Low. V3 is a superset of V2 deck so I don't anticipate things breaking if tests pass

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
No problems for ot3 for me - I guarantee we'll have to make many changes with the ot3 deck, so no need to worry.

File: shared-data/deck/definitions/3/ot3_standard.json [hunk 1]
```diff
@@ -146,7 +183,11 @@
           "zDimension": 0
         },
         "displayName": "Slot 11",
-        "compatibleModuleTypes": ["magneticModuleType", "temperatureModuleType"]
+        "compatibleModuleTypes": [
+          "magneticModuleType",
+          "temperatureModuleType",
+          "heaterShakerModuleType"
```
Comment (shlokamin) [inline shared-data/deck/definitions/3/ot3_standard.json:189]: can the heater shaker go in slot 11?

File: shared-data/deck/definitions/3/ot3_standard.json [hunk 2]
```diff
@@ -110,7 +135,11 @@
           "zDimension": 0
         },
         "displayName": "Slot 8",
-        "compatibleModuleTypes": ["magneticModuleType", "temperatureModuleType"]
+        "compatibleModuleTypes": [
+          "magneticModuleType",
+          "temperatureModuleType",
+          "heaterShakerModuleType"
```
Comment (shlokamin) [inline shared-data/deck/definitions/3/ot3_standard.json:141]: should the heater shaker be allowed to go in any of the middle slots?

Comment (sanni-t) [reply shared-data/deck/definitions/3/ot3_standard.json:141]: The heater shaker is not recommended to be placed in middle slots but if a user wants to, they can place it in the middle slot and still be able to use the module fully afaik. We're using the same reasoning for allowing other temp & mag module placements in the middle slots.

**Review by shlokamin — APPROVED** [review]
LGTM!

File: shared-data/deck/definitions/3/ot3_standard.json (hunk 1 above)
Comment (sanni-t) [reply shared-data/deck/definitions/3/ot3_standard.json:189]: Hmm.. the standard trash containers are too tall to be allowed next to the heater-shaker but I'm not sure if we want to call that a slot 11 incompatibility. We load trash containers at the beginning of a run so the user will/should get an error if they have a trash container loaded and they try to place the h/s next to it. But I'm thinking about super users who remove their trash housings..

**Review by mcous — APPROVED** [review]
LGTM! Diff'd 3.json with 2.json manually, and the only differences were the addition of `heaterShakerModuleType` and setting the allowed version to `3`.

Should we delete `shared-data/deck/schemas/2.json` and `shared-data/deck/schemas/1.json`, now that they're not used?

**Comment by sanni-t** [comment]
> LGTM! Diff'd 3.json with 2.json manually, and the only differences were the addition of `heaterShakerModuleType` and setting the allowed version to `3`.

Thanks for checking!

> Should we delete `shared-data/deck/schemas/2.json` and `shared-data/deck/schemas/1.json`, now that they're not used?

I prefer keeping them so we have some reference to the updates in schemas. Helps explain where the version '3' is from, since all other references to previous versions are removed. But not too attached to them either so can remove them if necessary.

**Comment by sanni-t** [comment]
Going to merge this so it unblocks the H/S in PAPI work. Will revisit if required when we have a decision about the slot 11 incompatibility.

**Pull request merged**